src/scripts/main.go

```diff
@@ -50,95 +52,58 @@ func main() {
 		}
 	}
 
-	// Exit if the job status does not match the message send event and the message send event is not set to "always"
-	if !stringutils.IsEventMatchingStatus(conf.EventToSendMessage, conf.JobStatus) {
-		message := fmt.Sprintf("The job status %q does not match the status set to send alerts %q.", conf.JobStatus, conf.EventToSendMessage)
-		fmt.Println(message)
-		fmt.Println("Exiting without posting to Slack...")
-		os.Exit(0)
-	}
-
-	// Check if the branch and tag match their respective patterns and parse the invert match parameter
-	branchMatches, err := stringutils.IsPatternMatchingString(conf.BranchPattern, conf.JobBranch)
-	if err != nil {
-		log.Fatal("Error parsing the branch pattern:", err)
-	}
-	tagMatches, err := stringutils.IsPatternMatchingString(conf.TagPattern, conf.JobTag)
-	if err != nil {
-		log.Fatal("Error parsing the tag pattern:", err)
-	}
 	invertMatch, _ := strconv.ParseBool(conf.InvertMatchStr)
-	if !stringutils.IsPostConditionMet(branchMatches, tagMatches, invertMatch) {
-		fmt.Println("The post condition is not met. Neither the branch nor the tag matches the pattern or the match is inverted.")
-		fmt.Println("Exiting without posting to Slack...")
-		os.Exit(0)
+	ignoreErrors, _ := strconv.ParseBool(conf.IgnoreErrorsStr)
+	channels := strings.Split(conf.ChannelsStr, ",")
+
+	slackNotification := slack.Notification{
+		Status:                   conf.JobStatus,
+		Branch:                   conf.JobBranch,
+		Tag:                      conf.JobTag,
+		Event:                    conf.EventToSendMessage,
+		BranchPattern:            conf.BranchPattern,
+		TagPattern:               conf.TagPattern,
+		InvertMatch:              invertMatch,
+		InlineTemplate:           conf.InlineTemplate,
+		EnvVarContainingTemplate: conf.EnvVarContainingTemplate,
 	}
 
-	// Build the message body
-	template, err := jsonutils.DetermineTemplate(conf.InlineTemplate, conf.JobStatus, conf.EnvVarContainingTemplate)
+	modifiedJSON, err := slackNotification.BuildMessageBody()
 	if err != nil {
-		log.Fatalf("%v", err)
-	}
-	if template == "" {
-		log.Fatalf("the template %q is empty. Exiting without posting to Slack...", template)
+		log.Fatalf("failed to build message body: %v", err)
 	}
 
-	// Expand environment variables in the template
-	templateWithExpandedVars, err := jsonutils.ApplyFunctionToJSON(template, jsonutils.ExpandEnvVarsInInterface)
-	if err != nil {
-		log.Fatal(err)
+	client := slack.NewClient(slack.ClientOptions{SlackToken: secret.String(conf.AccessToken)})
+
+	if !slackNotification.IsEventMatchingStatus() {
```

Comment:
I think we can fold the notification checks into BuildMessageBody and surface issues in the returned `error` - thoughts?

Reply:
moved the checks into `BuildMessageBody` like you suggested! 👍